jujugui/static/gui/src/app/app.js

```diff
@@ -1905,10 +1905,10 @@ YUI.add('juju-gui', function(Y) {
       if (localStorage.getItem('environmentName')) {
         environmentName = localStorage.getItem('environmentName');
       }
-      // The config.js jujuEnvUUID trumps 'sandbox'; in any other
+      // The config.js jujuEnvUUID trumps when the GUI is embedded; in any other
       // situation, the provided environmentName should update the app's
       // jujuEnvUUID.
-      if (environmentName === 'sandbox') {
+      if (this.get('embedded')) {
```

Comment:
This should be `this.get('embedded`) && environmentName === 'sandbox'`; we get environmentName up above and if another name has been assigned we don't want to trump that.